TwilioVerifyDemo/TwilioVerifyDemo/Common/AppDelegate.swift

```diff
@@ -82,28 +86,60 @@ extension AppDelegate: UNUserNotificationCenterDelegate {
       return
     }
     
-    showChallenge(payload: payload)
+    handleChallengeApproval(with: payload)
   }
 }
 
 private extension AppDelegate {
+  func handleChallengeApproval(with payload: [String: Any]) {
+    guard let payloadData = PushChallengePayloadData(payload: payload) else {
+      return
+    }
+
+    if AppModel.factorsSilentyApproved[payloadData.factorSid] == true {
+      updateChallenge(with: .approved,
+                      notificationPayload: payload,
+                      factorSid: payloadData.factorSid,
+                      challengeSid: payloadData.challengeSid)
+    } else {
+      showChallenge(payload: payload)
+    }
+  }
+  
+  func updateChallenge(with status: ChallengeStatus,
+                       notificationPayload: [String: Any],
+                       factorSid: String,
+                       challengeSid: String) {
+    
+    let payload = UpdatePushChallengePayload(
+      factorSid: factorSid,
+      challengeSid: challengeSid,
+      status: status
+    )
+    
+    twilioVerify?.updateChallenge(withPayload: payload, success: { [weak self] in
+      self?.showChallenge(payload: notificationPayload)
+    }) { error in
+      print("Unable to silenty approve this challenge, details: \(error.errorMessage)")
```

Comment:
Let's make use of the Logger here:

```swift
Logger.shared.log(withLevel: .error, message: error.localizedDescription)
```

Reply:
`Logger` should be used only in the SDK. Please use another mechanism.